File: core/src/main/java/com/larryhsiao/nyx/core/attachments/RemovalAttachmentByJotId.java
package com.larryhsiao.nyx.core.attachments;

import com.silverhetch.clotho.Action;
import com.silverhetch.clotho.Source;

import java.sql.Connection;
import java.sql.PreparedStatement;

/**
 * Remove Attachments by given Jot id.
 */
public class RemovalAttachmentByJotId implements Action {
    private final Source<Connection> dbConn;
    private final long jotId;

    public RemovalAttachmentByJotId(Source<Connection> dbConn, long jotId) {
        this.dbConn = dbConn;
        this.jotId = jotId;
    }

    @Override
    public void fire() {
        try (PreparedStatement stmt = dbConn.value().prepareStatement(
            // language=H2
            "DELETE FROM attachments " +
                "WHERE JOT_ID=?;"
        )) {
            stmt.setLong(1, jotId);
            stmt.executeUpdate();
        } catch (Exception e) {
            throw new IllegalArgumentException(e);
        }
    }
}

File: core/src/main/java/com/larryhsiao/nyx/core/attachments/RemovalAttachmentByJotId.kt
package com.larryhsiao.nyx.core.attachments

import com.silverhetch.clotho.Action
import com.silverhetch.clotho.Source
import java.sql.Connection

/**
 * Remove Attachments by given Jot id.
 */
class RemovalAttachmentByJotId(
    private val dbConn: Source<Connection>,
    private val jotId: Long
) : Action {
    override fun fire() {
        try {
            dbConn.value().prepareStatement( // language=H2
                "UPDATE attachments " +
                    "SET DELETE = 1 , VERSION = VERSION + 1 " +
                    "WHERE JOT_ID=?1;"
            ).use { stmt ->
                stmt.setLong(1, jotId)
                stmt.executeUpdate()
            }
        } catch (e: Exception) {
            throw IllegalArgumentException(e)
        }
    }
}